### PR TITLE
Add python3 virtual environment for docker-ptf

### PR DESCRIPTION
**Pull request by ZhaohuiS**
#### Why I did it
Migrate ptftests script to python3, in order to do an incremental migration, add python virtual environment firstly, install all required python packages in virtual env as well.
Then migrate ptftests scripts from python2 to python3 one by one avoid impacting non-changed scripts.

Signed-off-by: Zhaohui Sun <zhaohuisun@microsoft.com>

#### How I did it
Add python3 virtual environment for docker-ptf. 
Add submodule `ptf-py3` and install patched ptf 0.9.3 into virtual environment as well, two ptf issues were reported here:
https://github.com/p4lang/ptf/issues/173
https://github.com/p4lang/ptf/issues/174

#### How to verify it
login docker-ptf container and check if there is a folder ` /root/env-python3/bin`

1. Connect to the docker-ptf container,  take kvm testbed as example as below:
```
$ docker exec -ti ptf_vms6-1 bash
```
1. Activate the virtual environment
```
$ source /root/env-python3/bin/activate
(env-python3) root@70d4f7e9dfcc:/# python
Python 3.7.3 (default, Jan 22 2021, 20:04:44) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.

>>> import ptf
```
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: dockers/docker-ptf/Dockerfile.j2 [hunk 1]
```diff
@@ -117,10 +131,62 @@ RUN rm -rf /debs \
...
+# Activating a virtualenv. The virtualenv automatically works for RUN, ENV and CMD.
+ENV VIRTUAL_ENV=/root/env-python3
+ARG BACKUP_OF_PATH="$PATH"
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+ENV LANG=C.UTF-8 LC_ALL=C.UTF-8 PYTHONIOENCODING=UTF-8
+
+RUN python3 -m pip install --upgrade --ignore-installed pip
+
+# Install all python modules from pypi. python-scapy is exception, ptf debian package requires python-scapy
```
Comment (wangxin) [inline dockers/docker-ptf/Dockerfile.j2:150]: In the comment, `python-scapy` -> `python3-scapy`?

File: dockers/docker-ptf/Dockerfile.j2 [hunk 2]
```diff
@@ -117,10 +131,62 @@ RUN rm -rf /debs \
...
+# TODO: Clean up this step
```
Comment (wangxin) [inline dockers/docker-ptf/Dockerfile.j2:151]: What to be cleaned up?

**Review by wangxin — APPROVED** [review]
LGTM. Just a few comments need improvements.

**Review by Pterosaur — COMMENTED** [review]
Should we combine `ptf-py3` and `ptf-py3.patch` to one folder,`ptf-py3`, or combine the existing `ptf` with them to a same folder `ptf`? I think that will look better to code structure.

**Comment by ZhaohuiS** [comment]
> 

Yes, exactly, I just followed the same rule as scapy patch in this PR #10457

File: dockers/docker-ptf/Dockerfile.j2 (hunk 1 above)
Comment (ZhaohuiS) [reply dockers/docker-ptf/Dockerfile.j2]: Done

File: dockers/docker-ptf/Dockerfile.j2 (hunk 2 above)
Comment (ZhaohuiS) [reply dockers/docker-ptf/Dockerfile.j2]: > What to be cleaned up?

Yes, remove this useless comment.

**Comment by ZhaohuiS** [comment]
@lguohan @qiluo-msft @xumia Could you please review and approve this change? Thank you.

File: dockers/docker-ptf/Dockerfile.j2 [hunk 3]
```diff
@@ -117,10 +131,88 @@ RUN rm -rf /debs \
...
+ENV LANG=C.UTF-8 LC_ALL=C.UTF-8 PYTHONIOENCODING=UTF-8
+
+RUN python3 -m pip install --upgrade --ignore-installed pip
+
+# Install all python modules from pypi. python3-scapy is exception, ptf debian package requires python3-scapy
+RUN python3 -m pip install setuptools \
+    && pip3 install supervisor \
+    && pip3 install ipython==5.4.1 \
+    && pip3 install Cython \
+    && git clone https://github.com/sflow/sflowtool \
```
Comment (lguohan) [inline dockers/docker-ptf/Dockerfile.j2:155]: python venv should be python, not sure if sflowtool, openbfdd, nanomsg has python part?

**Review by wangxin — APPROVED** [review]

**Pull request merged**